### PR TITLE
Fix #4093 - explain how -Debug changes preference variable

### LINKED ISSUE #4093: Wrong description
The documentation states: 

"The third command writes a debug message. It uses the Debug common parameter to override the value of $DebugPreference and to display the debug messages resulting from this command.
As a result, even though the value of $DebugPreference is SilentlyContinue, the debug message appears.". 

This is a bit misleading. The debug switch actually changes the $DebugPreference within the context of that command. Also to be seen in the following example: 

function testf {
    [cmdletbinding()]
    param()
    Write-Host "DebugPreference is $DebugPreference"
} 

PS C:\> testf
DebugPreference is SilentlyContinue
PS C:\> testf -debug
DebugPreference is Inquire
PS C:\>

Moreover, as seen in the above example, the -debug switch doesn't just "display" the message as is stated in the helptext - it sets $DebugPreference to "Inquire" (for some reason totally different from the verbose switch, which would set it to "continue"), and halts execution to inquire if execution should continue - which is very different when used in scripts, for example. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ef95204d-9964-c3fb-7e51-854736609b04
* Version Independent ID: 0604419e-8dc3-2a03-4641-cde87780c864
* Content: [Write-Debug (Microsoft.PowerShell.Utility)](https://docs.microsoft.com/en-us/powershell/module/Microsoft.PowerShell.Utility/Write-Debug?view=powershell-5.0)
* Content Source: [reference/5.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/5.0/Microsoft.PowerShell.Utility/Write-Debug.md)
* Product: **powershell**
* Technology: **powershell-cmdlets**
* GitHub Login: @JamesWTruher
* Microsoft Alias: **jimtru**

### PR DESCRIPTION
**Pull request by sdwheeler**
Version(s) of document impacted
------------------------------
- [ ] Impacts 6.next document
- [x] Impacts 6 document
- [x] Impacts 5.1 document
- [x] Impacts 5.0 document
- [x] Impacts 4.0 document
- [x] Impacts 3.0 document

Reason(s) for not updating all version of documents
--------------------------------------------------
- [ ] The documented feature was introduced in version (list version here) of PowerShell
- [ ] This issue only shows up in version (list version(s) here) of the document
- [ ] This PR partially fixes the issue, and issue #<insert here> tracks the remaining work

### CODE REVIEW

**Comment by zjalexander** [comment]
Docs Build status updates of commit _[fa26096](https://github.com/sdwheeler/PowerShell-Docs/pull/4116/commits/fa260968a6dbd41657c858ccf7da0125a9358199)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-3.0&branch=pr-en-us-4116) (powershell-3.0) |
[reference/4.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/4.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-4.0&branch=pr-en-us-4116) (powershell-4.0) |
[reference/5.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/5.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-5.0&branch=pr-en-us-4116) (powershell-5.0) |
[reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-5.1&branch=pr-en-us-4116) (powershell-5.1) |
[reference/6/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/6/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-6&branch=pr-en-us-4116) (powershell-6) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C4%5C6ff7e8c3-dfc6-3ebd-da5a-d5e2ff43de8f%5CPullRequest%5C201904041625416259-4116%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=BEteS3n%2B8lfb56KJpoq15%2Bt%2F1btK7Qvfz%2FJFZ%2BotxFA%3D&st=2019-04-04T16%3A25%3A25Z&se=2019-05-05T16%3A30%3A25Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by zjalexander** [comment]
Docs Build status updates of commit _[9159ea5](https://github.com/sdwheeler/PowerShell-Docs/pull/4116/commits/9159ea52c27c5b3aaa865b6fbd9e6170eb9f2796)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-3.0&branch=pr-en-us-4116) (powershell-3.0) |
[reference/4.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/4.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-4.0&branch=pr-en-us-4116) (powershell-4.0) |
[reference/5.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/5.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-5.0&branch=pr-en-us-4116) (powershell-5.0) |
[reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-5.1&branch=pr-en-us-4116) (powershell-5.1) |
[reference/6/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/6/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-6&branch=pr-en-us-4116) (powershell-6) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C4%5C6ff7e8c3-dfc6-3ebd-da5a-d5e2ff43de8f%5CPullRequest%5C201904041637564775-4116%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=KBBvaJaIDEprJr%2FxKU6OELsG%2B%2B5V%2Fp7K2PmvsIFuHz0%3D&st=2019-04-04T16%3A37%3A38Z&se=2019-05-05T16%3A42%3A38Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by sdwheeler** [comment]
#sign-off

File: reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-locale:  en-us
-keywords:  powershell,cmdlet
-online version:  http://go.microsoft.com/fwlink/?LinkID=113424
-external help file:  Microsoft.PowerShell.Commands.Utility.dll-Help.xml
-title:  Write-Debug
+external help file: Microsoft.PowerShell.Commands.Utility.dll-Help.xml
+keywords: powershell,cmdlet
+locale: en-us
+Module Name: Microsoft.PowerShell.Utility
+ms.date: 06/09/2017
```
Comment (DCtheGeek) [inline reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md]: Enough content updated to trigger date change.

File: reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md [hunk 2]
```diff
@@ -21,98 +21,79 @@ Write-Debug [-Message] <String> [<CommonParameters>]
...
+This example shows the effect of changing the value of the `$DebugPreference` variable. First, we
+display the current value of `$DebugPreference` and attempt to write a debug message. The we change
+the value of `$DebugPreference` to **Continue**, which allows debug messages to be displayed.
 
-```powershell
-Write-Debug "Cannot open file." -debug
 ```
-
-```output
+PS C:\> $DebugPreference
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md]: fixed

File: reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md (hunk 1 above)
Comment (sdwheeler) [reply reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md]: Fixed

File: reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md [hunk 3]
```diff
@@ -21,98 +21,79 @@ Write-Debug [-Message] <String> [<CommonParameters>]
...
-```powershell
-Write-Debug "Cannot open file."
-```
+### Example 2: Change the value of $DebugPreference
 
-```output
-PS>
-```
+This example shows the effect of changing the value of the `$DebugPreference` variable. First, we
+display the current value of `$DebugPreference` and attempt to write a debug message. The we change
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md]: fixed

**Comment by zjalexander** [comment]
Docs Build status updates of commit _[7751f6d](https://github.com/sdwheeler/PowerShell-Docs/pull/4116/commits/7751f6d24b494a1922877fc00196db57004fc673)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-3.0&branch=pr-en-us-4116) (powershell-3.0) |
[reference/4.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/4.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-4.0&branch=pr-en-us-4116) (powershell-4.0) |
[reference/5.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/5.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-5.0&branch=pr-en-us-4116) (powershell-5.0) |
[reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-5.1&branch=pr-en-us-4116) (powershell-5.1) |
[reference/6/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/6/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-6&branch=pr-en-us-4116) (powershell-6) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C4%5C6ff7e8c3-dfc6-3ebd-da5a-d5e2ff43de8f%5CPullRequest%5C201904041707189646-4116%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=HL0NqT2wn70iHLrAGh%2FaEUHbjYeIZgEfFjfsnTt%2Fr%2Bo%3D&st=2019-04-04T17%3A07%3A22Z&se=2019-05-05T17%3A12%3A22Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (DCtheGeek) [reply reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md]: Not fixed.  I still see many `PS C:\>` prompts.

**Comment by zjalexander** [comment]
Docs Build status updates of commit _[9d2ff90](https://github.com/sdwheeler/PowerShell-Docs/pull/4116/commits/9d2ff90b4913f14cfaa6e35487553336c504a0d9)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-3.0&branch=pr-en-us-4116) (powershell-3.0) |
[reference/4.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/4.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-4.0&branch=pr-en-us-4116) (powershell-4.0) |
[reference/5.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/5.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-5.0&branch=pr-en-us-4116) (powershell-5.0) |
[reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-5.1&branch=pr-en-us-4116) (powershell-5.1) |
[reference/6/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/6/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-6&branch=pr-en-us-4116) (powershell-6) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C4%5C6ff7e8c3-dfc6-3ebd-da5a-d5e2ff43de8f%5CPullRequest%5C201904041746097025-4116%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=VwXl%2BviLVAzt9Cp0%2BTovOpq6Z5iprcclPRXlHzNeY9k%3D&st=2019-04-04T17%3A46%3A28Z&se=2019-05-05T17%3A51%3A28Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by zjalexander** [comment]
Docs Build status updates of commit _[ff3d86d](https://github.com/sdwheeler/PowerShell-Docs/pull/4116/commits/ff3d86d74b143055df6ffeac9821749c59765c11)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/3.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-3.0&branch=pr-en-us-4116) (powershell-3.0) |
[reference/4.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/4.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-4.0&branch=pr-en-us-4116) (powershell-4.0) |
[reference/5.0/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/5.0/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-5.0&branch=pr-en-us-4116) (powershell-5.0) |
[reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-5.1&branch=pr-en-us-4116) (powershell-5.1) |
[reference/6/Microsoft.PowerShell.Utility/Write-Debug.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-issue4093/reference/6/Microsoft.PowerShell.Utility/Write-Debug.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/PowerShell/module/microsoft.powershell.utility/write-debug?view=powershell-6&branch=pr-en-us-4116) (powershell-6) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C4%5C4%5C6ff7e8c3-dfc6-3ebd-da5a-d5e2ff43de8f%5CPullRequest%5C201904041800487248-4116%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=ePSS%2BvOCZknOI0x1xXUDU%2Bc1zgbYH8wNuPmPyBjyCB8%3D&st=2019-04-04T18%3A00%3A37Z&se=2019-05-05T18%3A05%3A37Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**